libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts

```diff
@@ -193,9 +194,32 @@ export class ResourceExplorer {
     }
 
     /**
-     * Build type for given $kind.
-     * @param kind $kind.
-     * @param config Source configuration object.
+     * Register a declarative type with the resource loader system.
+     *
+     * @template T The type of object.
+     * @param {string} kind The $kind name to map to this type.
+     * @param {new (...args: ?[]) => T} type Type of object to create.
+     * @param {CustomDeserializer}  loader Optional custom deserializer.
+     * @returns {ResourceExplorer} Resource explorer for fluent style multiple calls.
+     */
+    public registerType<T>(
+        kind: string,
+        type: new (...args: unknown[]) => T,
+        loader?: CustomDeserializer<T, Record<string, unknown>>
+    ): ResourceExplorer {
+        this.registerComponentTypes();
+        this.registerTypeInternal<T, Record<string, unknown>>(kind, type, loader);
```

Comment:
The type parameters might be able to be inferred here, not sure you need to list them explicitly.